### PR TITLE
grc: start process on flowgraph folder

### PR DESCRIPTION
**Pull request by japm48**
Allow the use of predictable relative paths.

This also includes changes on cmake compilation.

### CODE REVIEW

**Comment by mormj** [comment]
Love the idea of predictable relative paths - tagging a few more devs just to get eyes on this since it potentially changes how existing flowgraphs run

File: grc/gui/Executor.py [hunk 1]
```diff
@@ -83,11 +87,14 @@ def _cpp_popen(self):
 
         xterm_executable = find_executable(self.xterm_executable)
 
-        run_command_args = ['cmake .. &&', 'make && ', xterm_executable, '-e', run_command]
-        Messages.send_start_exec(' '.join(run_command_args))
+        # See: https://docs.python.org/3.8/library/os.html#os.cpu_count
+        nproc = len(os.sched_getaffinity(0))
```
Comment (michaelld) [inline grc/gui/Executor.py:91]: not surprisingly: `os.sched_getaffinity` is not defined on OSX ... Python's build system checks for whether `sched_setaffinity` exists and works on the target OS, and if not then the `os` module is built without support for it!

You can always use something like `min(int(os.cpu_count()/2)-1,1)` ...

Comment (japm48) [reply grc/gui/Executor.py:91]: Thanks, also broken on Windows...
Why half cpu_count, though?
Note that this only affects compilation, not flowgraph runtime.

Comment (michaelld) [reply grc/gui/Executor.py:91]: For some reason, `cpu_count` returns the number of hyperthreads, not cores, at least on my OSX laptop. We really don't want to use that many compile threads! 1 per actual core max, and preferable 1 smaller than that, provides a reasonable trade-off between compile time and CPU saturation.

Comment (japm48) [reply grc/gui/Executor.py]: Ok. I could not find a portable way to find the physical cores, so I basically used your solution.

**Comment by michaelld** [comment]
All CI passes. Multiple positive reviews. Merging.

**Pull request merged**